addons/mail/static/src/messaging/component/thread_needaction_preview/thread_needaction_preview.js

```diff
@@ -0,0 +1,118 @@
+odoo.define('mail.messaging.component.ThreadNeedactionPreview', function (require) {
+'use strict';
+
+const components = {
+    MessageAuthorPrefix: require('mail.messaging.component.MessageAuthorPrefix'),
+    PartnerImStatusIcon: require('mail.messaging.component.PartnerImStatusIcon'),
+};
+const useStore = require('mail.messaging.component_hook.useStore');
+const mailUtils = require('mail.utils');
+
+const { Component } = owl;
+
+class ThreadNeedactionPreview extends Component {
+
+    /**
+     * @override
+     */
+    constructor(...args) {
+        super(...args);
+        useStore(props => {
+            const thread = this.env.entities.Thread.get(props.threadLocalId);
+            const mainThreadCache = thread ? thread.mainCache : undefined;
+            let lastNeedactionMessageAuthor;
+            let lastNeedactionMessage;
+            if (thread) {
+                lastNeedactionMessage = mainThreadCache.lastNeedactionMessage;
+            }
+            if (lastNeedactionMessage) {
+                lastNeedactionMessageAuthor = lastNeedactionMessage.author;
+            }
+            return {
+                isDeviceMobile: this.env.messaging.device.isMobile,
+                lastNeedactionMessage: lastNeedactionMessage ? lastNeedactionMessage.__state : undefined,
+                lastNeedactionMessageAuthor: lastNeedactionMessageAuthor
+                    ? lastNeedactionMessageAuthor.__state
+                    : undefined,
+                thread: thread ? thread.__state : undefined,
+                threadCorrespondent: thread && thread.correspondent
+                    ? thread.correspondent.__state
+                    : undefined,
```

Comment:
Make `threadCorrespondent` at top for consistency, and just keep the "state" check here